### ISSUE TITLE
[Perf] MOSS TTS: batched vocoder decode

### ISSUE THREAD

**Issue by Ccyest** [sgl-project/sglang-omni#738]
## Motivation

The MOSS TTS vocoder stage (`create_vocoder_executor` in `sglang_omni/models/moss_tts/stages.py`) has a `batch_compute_fn` wired up, but it is a **serial loop** — not a true batched GPU decode:

```python
def _vocode_batch(payloads: list[StagePayload]) -> list[StagePayload]:
    return [_vocode(payload) for payload in payloads]
```

Each `_vocode()` call runs `processor.decode_audio_codes([segment])` per-segment, per-request. At low batch sizes the GPU is severely underutilized because the ConvTranspose decoder kernels are memory-bound and their launch overhead dominates.

The underlying audio tokenizer likely supports batched decode input. Padding variable-length code sequences and running a single batched decode should reduce per-request vocoder latency roughly proportional to the batch size.

## Current path

```python
def _decode_audio(state, delayed_codes):
    segments = split_moss_audio_segments(delayed_codes, ...)
    decoded = []
    for segment in segments:
        decoded.extend(processor.decode_audio_codes([segment]))  # one segment at a time
    waveform = torch.cat([...], dim=0)
    return waveform, sample_rate
```

`SimpleScheduler` collects up to `max_batch_size=8` requests, but the `_vocode_batch` just loops serially.

## Proposed change

1. Implement true batched decode: collect all segments across all requests in the batch, pad to uniform length, run a single batched `processor.decode_audio_codes()`, then trim padding and redistribute waveforms back to individual requests.
2. Use length-bucketing (like the encoder) if the decoder is sensitive to zero-padding artifacts.

## Acceptance Criteria

**Correctness**
- [ ] Batched decode produces identical waveforms to serial per-segment decode
- [ ] Multi-segment requests handled correctly (segments from same request reassembled in order)
- [ ] Existing unit tests still pass

**Performance**
- [ ] Vocoder throughput strictly higher than serial baseline at concurrency ≥ 2
- [ ] No regression at concurrency = 1

## Known Pitfalls

- **Variable segment lengths.** MOSS uses `split_moss_audio_segments()` to extract contiguous non-padding regions from the delay-pattern output. Different requests (and different segments within a request) will have different lengths.
- **Padding artifacts.** If the vocoder decoder uses non-causal convolutions, zero-padding may corrupt boundary samples. Trim carefully by codec frame count × hop size, and A/B compare against serial decode.
- **Multi-segment requests.** A single MOSS TTS request can produce multiple audio segments. The batched decode must track which segments belong to which request for correct reassembly.

## References

- Higgs batched vocoder: #569, PR #574
- Current MOSS vocoder: `sglang_omni/models/moss_tts/stages.py` lines 301–394
- Existing PR: #651 @yijiangt
- Tracking issue: #637

**Comment by Ccyest** [comment]
Closing — this is a MOSS-TTS **Delay**-only optimization, and we've moved our optimization focus to MOSS-TTS Local (v1.5).

For reference:
- **Delay (`moss_tts`)** still has the serial pseudo-batch noted in this issue: `_vocode_batch` just loops `[_vocode(p) for p in payloads]` (`sglang_omni/models/moss_tts/stages.py:386`). This is the same line as PR #651.
- **Local (`moss_tts_local`)** — a different architecture (Local Transformer, not the delay pattern) — already does genuine batched GPU decode: a single `processor.decode_audio_codes(codes_list)` over the whole batch, then redistributes (`sglang_omni/models/moss_tts_local/stages.py:424-439`). So this issue's goal is already met natively there.

Since Delay is no longer an optimization target and Local already covers batched vocoder decode, there's nothing left to do here. Closing as not planned.
